### ISSUE TITLE
Builder is unable to upgrade Asian plantation to level 2

### ISSUE THREAD

**Issue by abadger** [ldtteam/minecolonies#7444]
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.16.5
- MineColonies Version: 0.14.312-ALPHA
- Structurize Version: 0.13.204-ALPHA
- Related Mods and their Versions: Started off as a direwolf20-1.6 pack now with minecolonies, structurize, rats, tetra upgraded.

## Expected behavior

When the builder is asked to upgrade the asian plantation to level 2, he would do it.

## Actual behavior

The builder will request resources.  Then go to the plantation.  He walks up to the middle of the side of the plantation, tuns around, and comes back. (middle == between the hut and the fenced in sand area).  If I put up barriers so that he has to approach from the front of the plantation, he walks to the middle of the plantation, turns around, and then comes back.

## Steps to reproduce the issue

1. Have a level 1 Asian plantation
2. I have a level 5 asian builder
3. Select Build Options::Upgrade on the plantation.
4. Provide the builder with required resources until he makes the journey to the plantation.
5. Follow the builder and notice that he walks to the plantation and then immediately turns around and comes back without placing anything.

### Logs

- latest.log: https://gist.github.com/abadger/9daddd20400258d0211fd8b0341b452f
- crashlog: 

### Notes
Things I've tried without success:
* I've tried removing the grass paths.
* I've tried giving the builder all required resources, cancelling the upgrade, restarting minecraft, and then starting the upgrade again.

Possible culprit?  I notice that the Asian plantation has stairs underwater.  Could it be that the builder has problems dealing with the minecraft waterlogging mechanic? 

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by abadger** [comment]
I manually placed and waterlogged all the stairs on the lowest level.  The builder then proceeded to build the rest of the plantation (including waterlogged cobblestone stairs on the second-to-lowest y level).  My guess is that waterlogged mossy stone brick stairs are the issue.

**Comment by Raycoms** [comment]
Okay, I can't reproduce this here atm, did you rotate or mirror the hut? Which iteration style did you use?

**Comment by abadger** [comment]
Looks like I did not mirror it.  I probably did rotate it.although I can't verify for sure.  I'm not sure what an iteration style is.

**Comment by Raycoms** [comment]
Did you change any structurize settings? Or are you playing on a server?

**Comment by abadger** [comment]
I'm playing single player.  I did not change any structurize settings.

**Comment by abadger** [comment]
I built it over an area that had pools of water already.  I don't think that when I placed my mossy stone brick stairs that any of those had water underneath them, though.

**Comment by abadger** [comment]
Okay, it looks like the water underneath the asian plantation (or perhaps just a non-solid block?) is a factor.  In a superflat test world, I first built a plantation on solid ground and upgraded it to level 2 successfully.  Then I did the following which reproduced the issue:

* Dig a hole two deep for the whole footprint of an asian plantation.
* Fill the bottom with a layer of water source blocks.
* Build your asian plantation so that the cobblestone border is one layer above the ground surface.
* When the asian plantation is upgraded to level 2, the builder will get stuck moving back and forth between the builder's hut and the job site.

Picture of the plantation showing its level relative to the ground's surface:
![2021-08-01_15 49 10](https://user-images.githubusercontent.com/209242/127787628-602547be-8e49-45bd-933c-3cb2a29469b1.png)

Picture showing the layer of water underneath the plantation:
![2021-08-01_15 49 25](https://user-images.githubusercontent.com/209242/127787631-e7a8a973-af28-4cf2-ab55-dbc8d136b5ee.png)
